pkg/nar/dump.go

```diff
@@ -8,9 +8,20 @@ import (
 	"syscall"
 )
 
+// SourceFilterFunc is the interface for creating source filters,
+// it mimics the behaviour of the Nix builtin filterSource.
```

Comment:
Can you describe behaviour in a bit more detail?

 - Keep on true/omit on false
 - The input values into the filter function. I assume `path` would be a more fitting name for the first argument?